Sofa/Component/SolidMechanics/FEM/HyperElastic/src/sofa/component/solidmechanics/fem/hyperelastic/TetrahedronHyperelasticityFEMDrawing.h

```diff
@@ -38,9 +38,10 @@ void drawHyperelasticTets(const core::visual::VisualParams* vparams, const VecCo
         p.reserve(3 * topology->getNbTetrahedra());
     }
 
-    for(core::topology::Topology::TetrahedronID i = 0 ; i < topology->getNbTetrahedra();++i)
+    const auto& tetraArray = topology->getTetrahedra();
+    for(core::topology::Topology::TetrahedronID i = 0 ; i < tetraArray.size();++i)
     {
-        const auto t = topology->getTetrahedron(i);
+        const auto t = tetraArray[i];
```

Comment:
(cannot suggest)
No point to get the index and t is making an unnecessary copy.
Suggestion:
```
for(const auto& t : topology->getTetrahedra())
{
...
```